File: BOJ/15000-16000/15651/15651.py
N, M = map(int, input().split())


def print_set(arr, selected, toPick):
    if toPick == 0:
        print(*selected)
        return

    for i in range(N):
        print_set(arr, selected+[arr[i]], toPick-1)


print_set([i+1 for i in range(N)], [], M)

File: BOJ/15000-16000/15651/15651.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>

using namespace std;

int N, M;
int arr[] = {1, 2, 3, 4, 5, 6, 7, 8};

void print_set(vector<int> &selected, int toPick)
{
    if (toPick == 0)
    {
        for (int i = 0; i < selected.size(); i++)
            cout << selected[i] << " ";
        cout << "\n";
        return;
    }
    for (int i = 0; i < N; i++)
    {
        selected.push_back(arr[i]);
        print_set(selected, toPick - 1);
        selected.pop_back();
    }
}
int main()
{
    cin >> N >> M;
    vector<int> v;

    print_set(v, M);

    return 0;
}